### PR TITLE
fix(transport): restore support for GOOGLE_CLOUD_UNIVERSE_DOMAIN env var

### PR DESCRIPTION
**Pull request by quartzmo**
The `GOOGLE_CLOUD_UNIVERSE_DOMAIN `env variable no longer works.

If I use `google.golang.org/api@v0.191.0` I can start my example with `GOOGLE_CLOUD_UNIVERSE_DOMAIN=example.com` and it works without specifying `WithUniverseDomain` option explicitly.

When I use `google.golang.org/api@v0.197.0` and `cloud.google.com/go/auth@v0.9.4` I get error when I am using this env, if I don't specify `WithUniverseDomain` explicitly.

Error: the configured universe domain ("googleapis.com") does not match the universe domain found in the credentials ("example.com"). If you haven't configured the universe domain explicitly, "googleapis.com" is the default

### CODE REVIEW

File: transport/http/dial_test.go [hunk 1]
```diff
@@ -26,3 +31,21 @@ func TestNewClient(t *testing.T) {
 		t.Fatalf("got %s, want: %s", got, want)
 	}
 }
+
+func TestNewClient_NewAuthUniverseDomain(t *testing.T) {
+	os.Setenv("GOOGLE_CLOUD_UNIVERSE_DOMAIN", "example.com")
```
Comment (codyoss) [inline transport/http/dial_test.go:36]: nit: use https://pkg.go.dev/testing#T.Setenv. then you don't need to do this cleanup game.

File: transport/http/dial.go [hunk 2]
```diff
@@ -130,7 +133,7 @@ func newClientNewAuth(ctx context.Context, base http.RoundTripper, ds *internal.
 			DefaultScopes:           ds.DefaultScopes,
 			SkipValidation:          skipValidation,
 		},
-		UniverseDomain: ds.UniverseDomain,
+		UniverseDomain: ds.GetUniverseDomain(),
```
Comment (codyoss) [inline transport/http/dial.go:136]: Do we need to do this in the new library as well for those that use it directly?

Comment (quartzmo) [reply transport/http/dial.go:136]: Yes, `DialSettings.UniverseDomain` usages should be replaced anywhere they occur with `DialSettings.GetUniverseDomain()` in order to support the env var.

File: transport/http/dial_test.go (hunk 1 above)
Comment (quartzmo) [reply transport/http/dial_test.go]: Fixed, thanks!

File: transport/http/dial.go (hunk 2 above)
Comment (codyoss) [reply transport/http/dial.go:136]: The new lib does not have dialsettings, though. But it does allow someone to explicitly set this value. But if they don't, should we honor this there? If we did honor this in the new library and had the current code here I think the bug would be fixed for both, right?

Comment (quartzmo) [reply transport/http/dial.go:136]: I will update this in `grpc/dial.go` in this PR, thank you.

**Comment by quartzmo** [comment]
Closing in favor of a fix in the new `auth` module:

https://github.com/googleapis/google-cloud-go/pull/10915

**Pull request closed without merging**